modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -866,7 +918,12 @@ public String getName() {
         }
     };
     public final double getPosition() { return position.get(); }
-    public final void setPosition(double value) { position.set(value); }
+    public final void setPosition(double value) {
+        position.set(value);
+        // When the position is changed explicitly, we need to make sure
+        // the absolute offset is changed accordingly.
+        adjustAbsoluteOffset();
+    }
```

Comment:
This seems to violate the invariant `setPosition(...)` ≙ `positionProperty().set(...)`.

Your comment indicates that this is by design, but how do you prevent people from invoking `positionProperty().set(...)` and getting a different behavior?